libtalek/client.go

```diff
@@ -68,37 +55,63 @@ func NewClient(name string, config ClientConfig, leader common.LeaderInterface)
 
 /** PUBLIC METHODS (threadsafe) **/
 
+// SetConfig allows updating the configuraton of a Client, e.g. if server memebership
+// or speed characteristics for the system are changed.
 func (c *Client) SetConfig(config ClientConfig) {
 	c.config.Store(config)
 }
 
+// Kill stops client processing. This allows for graceful shutdown or suspension of requests.
 func (c *Client) Kill() {
 	atomic.StoreInt32(&c.dead, 1)
 }
 
+// Ping will perform an on-thread ping of the Talek system, allowing the client
+// to validate that it is connected to the Talek system, and check the latency
+// of connection to the server.
 func (c *Client) Ping() bool {
 	var reply common.PingReply
 	err := c.leader.Ping(&common.PingArgs{Msg: "PING"}, &reply)
 	if err == nil && reply.Err == "" {
 		c.log.Info.Printf("Ping success\n")
 		return true
-	} else {
-		c.log.Warn.Printf("Ping fail: err=%v, reply=%v\n", err, reply)
-		return false
 	}
+	c.log.Warn.Printf("Ping fail: err=%v, reply=%v\n", err, reply)
+	return false
+}
+
+// MaxLength returns the maximum allowed message the client can Publish.
+// TODO: support messages spanning multiple data items.
+func (c *Client) MaxLength() int {
+	config := c.config.Load().(ClientConfig)
+	return config.DataSize
 }
 
+// Publish a new message to the end of a topic.
 func (c *Client) Publish(handle *Topic, data []byte) error {
 	config := c.config.Load().(ClientConfig)
-	write_args, err := handle.GeneratePublish(config.CommonConfig, data)
+
+	if len(data) > config.DataSize-PublishingOverhead {
```

Comment:
Wouldn't it make more sense to have DataSize be the maximum size of any user-published data? As opposed to making the client keep track of what this overhead is.